apps/libsignpost/port_signpost.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+/* This is a port interface for the signpost API and networking stack!
+The goal is to export the minimal interface such that platforms
+can integrate with the signpost.
+
+From a high level you will need to provide:
+    - Blocking I2C write function (currently at least 255 bytes at a time)
+    - Asynchronous I2C slave function (listen on an address and send
+                                        back written bytes asynchronously)
+    - Set and clear of the MOD_OUT pin
+    - Asynchronous callback for falling edge of the Mod-in pin
+    - Currently the signpost libraries assume printf functionality
+        - This is assumed to be provided through <stdio.h>
```

Comment:
Why assume printf through stdio? You could conceptually have the signpost libraries snprintf to a buffer which is passed into a `void port_signpost_debug_print(char* msg)` function. That would allow platforms like arduino to implement different message printing functionality, or for some platforms to just drop print messages if they don't have a decent way of handling them.